src/gluonts/dataset/loader.py

```diff
@@ -11,186 +11,270 @@
+            self.manager.Event() for _ in range(self.num_workers)
+        ]
+        self.processes = []
 
+        for wid, event in enumerate(self.exhausted_events):
+            p = Process(
+                target=self.worker_fn,
+                args=(
+                    wid,
+                    self.num_workers,
+                    self.base_iterable,
+                    self.batch_size,
+                    self.batchify_fn,
+                    self.data_queue,
+                    self.terminate_event,
+                    event,
+                ),
+            )
+            p.start()
+            self.processes.append(p)
 
-class TrainDataLoader(DataLoader):
-    """
-    An Iterable type for iterating and transforming a dataset, in batches of a
-    prescribed size, until a given number of batches is reached.
-
-    The transformation are applied with in training mode, i.e. with the flag `is_train = True`.
-
-    Parameters
-    ----------
-    dataset
-        The dataset from which to load data.
-    transform
-        A transformation to apply to each entry in the dataset.
-    batch_size
-        The size of the batches to emit.
-    ctx
-        MXNet context to use to store data.
-    num_batches_per_epoch
-        Number of batches to return in one complete iteration over this object.
-    num_workers
-        The number of multiprocessing workers to use for data preprocessing.
-        By default 0, in which case no multiprocessing will be utilized.
-    num_prefetch
-        The number of prefetching batches only works if `num_workers` > 0.
-        If `prefetch` > 0, it allow worker process to prefetch certain batches before
-        acquiring data from iterators.
-        Note that using large prefetching batch will provide smoother bootstrapping performance,
-        but will consume more shared_memory. Using smaller number may forfeit the purpose of using
-        multiple worker processes, try reduce `num_workers` in this case.
-        By default `num_workers * 2`.
-    dtype
-        Floating point type to use. Default is np.float32.
-    shuffle_buffer_length
-        The length of the buffer used to do pseudo shuffle.
-        If not None, the loader will perform pseudo shuffle when generating batches.
-        Note that using a larger buffer will provide more randomized batches, but will make the job require a bit
-        more time to be done.
-    """
+        self.count = 0
+
+    @staticmethod
+    def worker_fn(
+        worker_id: int,
+        num_workers: int,
+        iterable: Iterable,
+        batch_size: int,
+        batchify_fn: Callable,
+        data_queue: Queue,
+        terminate_event,
+        exhausted_event,
+    ):
+        MPWorkerInfo.worker_process = True
+        MPWorkerInfo.worker_id = worker_id
+        MPWorkerInfo.num_workers = num_workers
+
+        for batch_list in batcher(iterable, batch_size):
+            batch = batchify_fn(batch_list)
```

Comment:
Don't like `batch_list`. Can we name `batch_list` -> `batch` and then `batch` -> `stacked_batch`?